File: dagger-mvp/src/main/java/be/vergauwen/simon/daggermvp/MVPApplication.java
package be.vergauwen.simon.daggermvp;

import android.app.Application;
import be.vergauwen.simon.daggermvp.di.ApplicationComponent;
import be.vergauwen.simon.daggermvp.di.ApplicationModule;
import be.vergauwen.simon.daggermvp.di.DaggerApplicationComponent;

public class MVPApplication extends Application {

    private static ApplicationComponent component;

    @Override
    public void onCreate() {
        super.onCreate();

        component = DaggerApplicationComponent.builder()
            .applicationModule(new ApplicationModule(this))
            .build();
    }

    public static ApplicationComponent getComponent() {
        return component;
    }
}

File: kotlin-dagger-mvp/src/main/kotlin/be/vergauwen/simon/kotlindaggermvp/MVPApplication.kt
package be.vergauwen.simon.kotlindaggermvp

import android.app.Application
import be.vergauwen.simon.kotlindaggermvp.di.ApplicationComponent
import be.vergauwen.simon.kotlindaggermvp.di.ApplicationModule
import be.vergauwen.simon.kotlindaggermvp.di.DaggerApplicationComponent

internal class MVPApplication : Application() {

  companion object {
    @JvmStatic lateinit var component: ApplicationComponent
  }

  override fun onCreate() {
    super.onCreate()

    component = DaggerApplicationComponent.builder()
        .applicationModule(ApplicationModule(this))
        .build()
  }
}